compiler/rustc_parse_format/src/lib.rs

```diff
@@ -813,101 +813,101 @@ fn find_skips_from_snippet(
     snippet: Option<string::String>,
     str_style: Option<usize>,
 ) -> (Vec<usize>, bool) {
+    if str_style.is_some() {
+        return (vec![], true);
+    }
+
     let snippet = match snippet {
         Some(ref s) if s.starts_with('"') || s.starts_with("r\"") || s.starts_with("r#") => s,
         _ => return (vec![], false),
     };
 
-    fn find_skips(snippet: &str, is_raw: bool) -> Vec<usize> {
-        let mut s = snippet.char_indices();
-        let mut skips = vec![];
-        while let Some((pos, c)) = s.next() {
-            match (c, s.clone().next()) {
-                // skip whitespace and empty lines ending in '\\'
-                ('\\', Some((next_pos, '\n'))) if !is_raw => {
-                    skips.push(pos);
-                    skips.push(next_pos);
-                    let _ = s.next();
-
-                    while let Some((pos, c)) = s.clone().next() {
-                        if matches!(c, ' ' | '\n' | '\t') {
-                            skips.push(pos);
-                            let _ = s.next();
-                        } else {
-                            break;
-                        }
-                    }
-                }
-                ('\\', Some((next_pos, 'n' | 't' | 'r' | '0' | '\\' | '\'' | '\"'))) => {
-                    skips.push(next_pos);
-                    let _ = s.next();
-                }
-                ('\\', Some((_, 'x'))) if !is_raw => {
-                    for _ in 0..3 {
-                        // consume `\xAB` literal
-                        if let Some((pos, _)) = s.next() {
-                            skips.push(pos);
-                        } else {
-                            break;
-                        }
+    let r_start = str_style.map_or(0, |r| r + 1);
+    let r_end = str_style.unwrap_or(0);
+    let snippet = &snippet[r_start + 1..snippet.len() - r_end - 1];
```

Comment:
```suggestion
    // The first and last characters are the double quotes.
    let snippet = &snippet[1..snippet.len() - 1];
```
We know that `str_style` is `None`.